websearch/websearch/src/event_source/mod.rs

```diff
@@ -0,0 +1,191 @@
+pub mod error;
+mod event_stream;
+mod ndjson_stream;
+mod parser;
+pub mod stream;
+pub mod types;
+mod utf8_stream;
+use crate::event_source::event_stream::SseWebsearchStream;
+use crate::event_source::stream::WebsearchStream;
+use crate::event_source::types::WebsearchStreamEntry;
+pub use error::StreamError;
+use golem_rust::wasm_rpc::Pollable;
+pub use ndjson_stream::NdJsonWebsearchStream;
+pub use parser::{is_bom, is_lf, line, RawEventLine};
+use reqwest::header::HeaderValue;
+use reqwest::{Response, StatusCode};
+use std::error::Error as StdError;
+use std::task::Poll;
+pub use stream::StreamType;
+pub use types::{
+    ImageResult, RateLimitInfo, SafeSearchLevel, SearchMetadata, SearchResult, StreamEnd,
+};
+pub use utf8_stream::Utf8Stream;
+
+/// Represents connection state of an [`EventSource`]
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+#[repr(u8)]
+pub enum ReadyState {
```

Comment:
Please remove the eventsource module and build the provider-specific SearchSessions directly.

This is overcomplicating things and leads to a lot of unimplemented! in the provider-specific code.
The search sessions should not be much more than some structs with interior mutability that hold the current page and advance it when requesting the next one.

Something like what you did for Tavily is fine, please adjust the other ones.